l5kit/l5kit/sampling/agent_sampling.py

```diff
@@ -80,16 +80,25 @@ def generate_agent_sample(
         target for that state. If you sample near the end of a scene, this may contain zeroes.
 
     """
-    agent_index_start = frames[0]["agent_index_interval"][0]
-    frames["agent_index_interval"] -= agent_index_start  # sync interval with the agents array
-
     #  the history slice is ordered starting from the latest frame and goes backward in time., ex. slice(100, 91, -2)
     history_slice = get_history_slice(state_index, history_num_frames, history_step_size, include_current_state=True)
-    history_frames = frames[history_slice]
-    history_agents = filter_agents_by_frames(history_frames, agents)
-
     future_slice = get_future_slice(state_index, future_num_frames, future_step_size)
-    future_frames = frames[future_slice]
+
+    history_frames = frames[history_slice].copy()  # copy() required if the object is a np.ndarray
+    future_frames = frames[future_slice].copy()
```

Comment:
One option to avoid copy is to pass the agent_index_offset `min_agents_index` to the function `filter_agents_by_frames`

Reply:
that's true. However, that would leave the indices untouched in the rest of the function. Right now those are not used, but if you stop in debug is better to have them aligned imo. But I'm absolutely available to change it if you have a strong opinion on that @yye-lyft . I don't think it would save us a lot, given that the number of picked frames  is low compared with that of the agents

Reply:
@lucabergamini agree, if the copy does not affect the performance I am okay with it. Yeah, better to have frame & agents aligned

Reply:
cpu utils seems to be very low with this update, so I think we should be fine. If anything come up in the future regarding this, we can always implement that